services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/RequestBatchManager.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.sqs.internal.batchmanager;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.services.sqs.batchmanager.BatchOverrideConfiguration;
+import software.amazon.awssdk.utils.Validate;
+
+@SdkInternalApi
+public abstract class RequestBatchManager<RequestT, ResponseT, BatchResponseT> {
+    private final int maxBatchItems;
+    private final Duration maxBatchOpenInMs;
+    private final BatchingMap<RequestT, ResponseT> requestsAndResponsesMaps;
+    private final BatchAndSend<RequestT, BatchResponseT> batchFunction;
+    private final BatchResponseMapper<BatchResponseT, ResponseT> responseMapper;
+    private final BatchKeyMapper<RequestT> batchKeyMapper;
+    private final ScheduledExecutorService scheduledExecutor;
+
+    protected RequestBatchManager(DefaultBuilder<RequestT, ResponseT, BatchResponseT, ?> builder) {
+        BatchConfiguration batchConfiguration = new BatchConfiguration(builder.overrideConfiguration);
+        this.requestsAndResponsesMaps = new BatchingMap<>(batchConfiguration.maxBatchKeys(),
+                                                          batchConfiguration.maxBufferSize(),
+                                                          RequestBatchBuffer::new);
+        this.maxBatchItems = batchConfiguration.maxBatchItems();
+        this.maxBatchOpenInMs = batchConfiguration.maxBatchOpenInMs();
+        this.batchFunction = Validate.notNull(builder.batchFunction, "Null batchFunction");
+        this.responseMapper = Validate.notNull(builder.responseMapper, "Null responseMapper");
+        this.batchKeyMapper = Validate.notNull(builder.batchKeyMapper, "Null batchKeyMapper");
+        this.scheduledExecutor = Validate.notNull(builder.scheduledExecutor, "Null scheduledExecutor");
```

Comment:
We shouldn't need these right? See comment above re: making these protected abstract methods

Reply:
yeap , removed these